### ISSUE TITLE
Problem exporting in Tiff/Jpeg file raw generated from Nikon D850

### ISSUE THREAD

**Issue by nikon-user** [darktable-org/darktable#6329]
**Describe the bug**
Darktable version 3.2.1
Platform: Windows 10
File Raw: NEF Nikon D850
File exported: Tiff/Jpeg

I correctly edit in Darktable file raw NEF generated from Nikon D850.
When exporting this file (CTRL-E), the file Tiff or Jpeg are partially or totally "black" (but not corrupted) and Darktable not crash.

The problem is present with all raw files generated from Nikon D850 and with various parameters setting for Tiff or Jpeg exported file.

Sometimes the export function freeze all PC and is necessary "hard-reset" (Windows 10 correctly updated).

No problem with:
- Raw NEF generated from Nikon D610 and Nikon D800
- Same raw file exported with previous version of Darktable 3.0.2.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Platform (please complete the following information):**
 - Darktable Version: 
 - OS: 
 - 
 - 

**Additional context**

**Comment by ptilopteri** [comment]
you need to anser:
- Is the issue still present using an empty/new config-dir?

**Comment by nikon-user** [comment]
> you need to anser:
> 
> * Is the issue still present using an empty/new config-dir?

Thanks for the reply

Now I'll made a new install (after removing all config file) of Darktable 3.2.1 on Windows 10.

**The bug persist** (view the attached file with Jpeg 60% compression).

![SSS_1925_01](https://user-images.githubusercontent.com/71519257/93663046-2190cc80-fa65-11ea-9a41-3c88219770dc.jpg)

**Comment by ptilopteri** [comment]
I regularly use the D850 and a Nikkor 600mm f/4 to shoot soccer and have never to my recollection had the problem you describe, but I am on openSUSE linux Tumbleweed, not windows and using darktable from git master.

Do you only observe the problem with a particular lense?
Have you tried to import an image from the D850 you worked successfully with an earlier version of darktable and not observed computer locking and "black" exported image?

Can you provide an unaltered nef and accompanying xmp file is problematic?

**Comment by nikon-user** [comment]
> Do you only observe the problem with a particular lense?
No. Various lens.

> Have you tried to import an image from the D850 you worked successfully with an earlier version of darktable and not observed computer locking and "black" exported image?
Yes. 
I have those results: 
All ok with  Darktable v**3.0.2** on Windows 10 / PC desktop with 12 GB ram.
All ok with  Darktable **v3.2.1 on Macos Catalina** / Macbook 13 year 2015.

**I think there is a default parameter  in Darktable for Windows v.3.2.1 (I made a fresh reinstall) that generate the problem only with this release.**

> Can you provide an unaltered nef and accompanying xmp file is problematic?
Yes, but the bug tracker is limited to 10 MB and the NEF file is 55MB.
If you give me an e-mail address, I'll try with WeTransfer service.

![DSC_0063](https://user-images.githubusercontent.com/71519257/93710354-4f9a0d80-fb46-11ea-877a-4d3906fd48d9.jpg)

The Export function is for me important because I use Darktable as RAW editor for GIMP and with this problem is impossible to operate.

Bye and thanks for your reply and help.

**Comment by nikon-user** [comment]
> Can you provide an unaltered nef and accompanying xmp file is problematic?
Here the link for NEF, xmp, generated file with Export:
https://drive.google.com/drive/folders/1jxzedz3-gl_59vmLut754CEs1_fBVY1x?usp=sharing

Same problem in **Darktable v.3.2.1 / Windows 10** with **all** files generated from Nikon D850 and exported to Jpeg (100% compression or not), TIFF (8/16/32 bit with or without compression), EXR.

With Darktable v.3.2.1 for Windows, **all OK with NEF generated from Nikon D610 or D800**.
The NEF in Nikon D850 are with 16 bit color and Nikon compression ON.

**Comment by ptilopteri** [comment]
The last image is not completely black but only partially exported.

looking at the images you provided.

**Comment by ptilopteri** [comment]
I imported you image into darktable git master b2995917b, then exported 
using your xmp, <ctrl><e> w/o any editing.

I also exported to gimp
exported image:  http://wahoo.no-ip.org/~paka/DSC_0063.jpg
screenshot of export to gimp:  
    http://wahoo.no-ip.org/~paka/Screenshot_20200920_082021.png

You apparently have something wrong with you local install or local build, or
the windows export from darktable into gimp.

Did you try directly exporting the image and then opening it in gimp?

Remember I am on openSUSE Timbleweed

**Comment by nikon-user** [comment]
> You apparently have something wrong with you local install or local build, or
> the windows export from darktable into gimp.

Also I think so, but **I am using the latest stable release 3.2.1 for Windows from fresh reinstall ** without modify the default parameters.

I' ll made another try with Linux and now I summarize the result:
- Latest stable release 3.2.1  from www.darktable.org
- Exporting in Jpeg/Tiff NEF file generated from Nikon D850

=> Darktable 3.2.1 / Windows 10 - file exported partially "blank".
=> Darktable 3.2.1 / Macos Catalina - file exported OK
=> Darktable 3.2.1 / Linux Ubuntu 20.04 (latest test) - file exported OK

> Did you try directly exporting the image and then opening it in gimp?
Either directly exporting the image in Gimp (file EXR) and also exporting in tiff as separated file. Same bug, partially blank.

At moment, your test is corresponding to mine (Linux is OK).
I think we have only two test for investigate this issue: 
- Another user with Nikon D850 and Windows 10.
- I could install another release (previous v.3.2.1) or the latest GIT development branch compiled for Windows.

For the latest test, **I need a link to a Darktable version compiled for Windows** (I cannot use the development tools)

**Comment by ptilopteri** [comment]
> Also I think so, but **I am using the latest stable release 3.2.1 for Windows from fresh reinstall ** without modify the default parameters.

did you remove the config files from windows?  They are hidden somewhere, appfiles or something like that.  I do not run windows.
 
> => Darktable 3.2.1 / Macos Catalina - file exported OK

would not that make you believe it was a windows build or environment issue?

> => Darktable 3.2.1 / Linux Ubuntu 20.04 (latest test) - file exported OK

again, windows build or environment issue?
 
> > Did you try directly exporting the image and then opening it in gimp?
> > Either directly exporting the image in Gimp (file EXR) and also exporting in tiff as separated file. Same bug, partially blank.

but *only* in windows?
 
> At moment, your test is corresponding to mine (Linux is OK).
> I think we have only two test for investigate this issue:
> 
>     * Another user with Nikon D850 and Windows 10.

make another user and try.
 
>     * I could install another release (previous v.3.2.1) or the latest GIT development branch compiled for Windows.

I would not worry about "another release".

> For the latest test, **I need a link to a Darktable version compiled for Windows** (I cannot use the development tools)

**Comment by nikon-user** [comment]
> did you remove the config files from windows? They are hidden somewhere, appfiles or something like that. I do not run windows.
> again, windows build or environment issue?

With yours suggestments, I have made some other tests:

A) With latest official release (v3.2.1), I have definied a new user profile on the same PC with Windows 10.
Same exporting malfunction: image partially "black".

B) I have installed Darktable v3.2.1 on a **new** PC with Windows 10.
BINGO **problem SOLVED**, image correctly exported.

As further investigation, on the first PC with Windows 10 (case A) I changed many parameters.
With these change in CPU / Memory section (Italian version), **problem SOLVED**.

![Parameters-Cpu-Memory-ok_screenshot](https://user-images.githubusercontent.com/71519257/93755674-fee8ea00-fc03-11ea-9d4d-1705a5a8b697.png)

These is the file correctly exported:
https://drive.google.com/drive/folders/1mnyrx6LfFS7hDdCVuq7MnkjfWi2fRwwJ?usp=sharing

**I suppose that the main problem in Windows is the Open CL acceleration enabled.** During many test i have detected some "freeze" of OS with OpenCL enabled.

Thank you for kindly support.

How close this issue on the bug-tracker ?
